packages/client/src/resources/run.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+
+import { NodeResponseStatus } from '~enums'
+import { CheckStatus, NodeRequest, ProjectFile } from '~interfaces'
+import {
+	checkEstimateStatus,
+	executeNode,
+	returnOrchestratorStatus,
+	terminateRun,
+	uploadFiles,
+} from '~resources'
+import { createFormData, isStatusProcessing, wait } from '~utils'
+import { buildLoadNode } from './prepareDoWhyData'
+
+export class Run {
+	private fileUrl = ''
+	private fileName = ''
+	public orchestratorResponse
+	private _onUpdate: ((status: CheckStatus) => void) | undefined
+	private _onComplete: ((status: CheckStatus) => void) | undefined
+	private _onCancel: (() => void) | undefined
+
+	constructor(
+		onUpdate?: (status: CheckStatus) => void,
```

Comment:
These would be better as an options object so any one could be omitted without needing to supply prior optional args.

E.g.: `
{
  onUpdate: (...) => void,
  onComplete: (...) => void,
  onCancel: (...) => void
}
`